### PR TITLE
FIND-81:detail page initial

### PR DESCRIPTION
**Pull request by TNA-Allan**
# Pull Request

Ticket URL: https://national-archives.atlassian.net/browse/FIND-81

## About these changes

As part of the transfer from https://github.com/nationalarchives/ds-wagtail

- Removed search detail path, view, template
- Added record detail page path with regex checks for id
- Added initial view, record details page template 
- Added 404 page
- Add tests

## How to check these changes

1) DEBUG=True (by default)
    http://localhost:65533/catalogue/id/C123/ --> record detail page

2) Set DEBUG=False (docker-compose.yml -> environment)
    http://localhost:65533/catalogue/id/Z123/ --> 404 page

## Before assigning to reviewer, please make sure you have

- Checked things thoroughly before handing over to reviewer
- Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- Ensured that PR includes only commits relevant to the ticket
- Waited for all CI jobs to pass before requesting a review
- Added/updated tests and documentation where relevant

### CODE REVIEW

**Review by JohnHeeryTNA — APPROVED** [review]
Approving from an FE perspective.
I've run the branch locally and can render the record detail page (DEBUG=True)
and also the 404 page (DEBUG=False).

Best if @Firdaussi reviews the .py code.

File: app/templates/404.html
Comment (Firdaussi) [inline app/templates/404.html:1]: I'm happy with 'page not found' as a placeholder, but should we at some point consider something with more information perhaps?

**Review by Firdaussi — COMMENTED** [review]
Just a couple of items.

**Comment by Firdaussi** [comment]
All issues resolved from the backend POV.

**Comment by Firdaussi** [comment]
All issues resolved.

**Review by Firdaussi — APPROVED** [review]
Approved

File: config/settings/develop.py [hunk 1]
```diff
@@ -11,6 +11,13 @@
 
 FORCE_HTTPS = strtobool(os.getenv("FORCE_HTTPS", "False"))
 
+DJANGO_SERVE_STATIC = True
+
+if not DEBUG and DJANGO_SERVE_STATIC:
```
Comment (ahosgood) [inline config/settings/develop.py:16]: `DJANGO_SERVE_STATIC` is always going to be true?

File: app/templates/records/record_detail.html [hunk 2]
```diff
@@ -1,12 +1,11 @@
 {% extends 'base.html' %}
 
-{%- set pageTitle = 'Catalogue item ' + id -%}
```
[the remark this answers is not in the document]
Comment (TNA-Allan) [reply app/templates/records/record_detail.html:3]: Restored `pageTitle` with `page_title`. Fix https://github.com/nationalarchives/ds-search/pull/9

**Pull request merged**